assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -2983,6 +2983,14 @@ public String getMaxPointGradeDisplay(int factor, int maxGradePoint) {
         return getGradeDisplay(Integer.toString(maxGradePoint), Assignment.GradeType.SCORE_GRADE_TYPE, factor);
     }
 
+    public String getFactoredGrade(int factor, String grade) {
+        String factoredGrade = "0,00";
+        if (grade != null){
+            factoredGrade = (grade.length() == 1? "0,0" + grade: (grade.length() == 2? "0," + grade : grade.substring(0, grade.length()-2) + "," + grade.substring(grade.length()-2, grade.length())));
```

Comment:
This looks a bit odd to me, could you explain it to me? AFAIK I see "0,00" as a potential grade but this will only work with some languages like ES.